### DISCUSSION TITLE
Is there an easy way to highlight/markup the matched tokens in the result documents returned by wink-bm25-text-search search method?

### DISCUSSION THREAD

**Discussion by bc2009** [winkjs/wink-nlp#131] [Help on coding]
for example, for the [example provided in the repo](https://github.com/winkjs/wink-bm25-text-search?tab=readme-ov-file#example-), is there a way to highlight the words `never`, `studied`, and `law` in the results? Looks like we can get the tokens by running `prepTask()` but these may not be the original words (like query has `not` but original is `never`).

**Comment by sanjayaksaxena** [comment]
Here are two immediate suggestions for addressing it:

1. If the 'negation flag' associated with the search term is set to true, then also highlight the immediately preceding non-negated token. The `its.negationFlag` helper of winkNLP will let you perform this check. However, please proceed with caution: although this approach is effective in most scenarios, there are instances where it might not succeed.

2. Or consider highlighting the entire sentence, or sentences, containing the search or query terms.  This can be accomplished using winkNLP's markup API.
